libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1063,6 +1063,64 @@ void TypeChecker::endVisit(EmitStatement const& _emit)
 	m_insideEmitStatement = false;
 }
 
+/**
+ * Creates a tuple declaration syntax from a vector of variable declarations.
+ *
+ * @param decls a tuple of variables
+ *
+ * @returns a Solidity language confirming string of a tuple variable declaration.
+ */
+static string createTupleDecl(vector<VariableDeclaration const*> const & decls)
+{
+	if (decls.size() == 1)
```

Comment:
Just a suggestion, which might make this easier to read and also less error prone:
```
vector<string> components;
for (auto const& decl: decls)
  if (decl)
    components.push_back(decl->annotation().type->toString(true) + " " + decl->name());
  else
    components.push_back({});
if (typeNames.size == 0)
  return typeNames.front();
else
  return "(" + boost::algorithm::join(typeNames, ", ") + ")";
```